src/System.Private.CoreLib/shared/System/Buffers/StandardFormat.cs

```diff
@@ -146,40 +146,47 @@ private static bool ParseHelper(ReadOnlySpan<char> format, out StandardFormat st
         /// </summary>
         public override string ToString()
         {
-            unsafe
+            Span<char> buffer = stackalloc char[FormatStringLength];
+            int charsWritten = Format(buffer);
+            return new string(buffer.Slice(0, charsWritten));
+        }
+
+        /// <summary>The length buffer needed by <see cref="Format"/>.</summary>
+        internal const int FormatStringLength = 4;
+
+        /// <summary>
+        /// Formats the format in classic .NET format.
+        /// </summary>
+        internal int Format(Span<char> destination)
+        {
+            Debug.Assert(destination.Length >= 4);
+
+            int count = 0;
+            char symbol = Symbol;
+            if (symbol != default)
             {
-                const int MaxLength = 4;
-                char* pBuffer = stackalloc char[MaxLength];
+                destination[count++] = symbol;
 
-                int dstIndex = 0;
-                char symbol = Symbol;
-                if (symbol != default)
+                byte precision = Precision;
+                if (precision != NoPrecision)
                 {
-                    pBuffer[dstIndex++] = symbol;
-
-                    byte precision = Precision;
-                    if (precision != NoPrecision)
+                    if (precision >= 100)
                     {
-                        if (precision >= 100)
-                        {
-                            pBuffer[dstIndex++] = (char)('0' + (precision / 100) % 10);
-                            precision = (byte)(precision % 100);
-                        }
-
-                        if (precision >= 10)
-                        {
-                            pBuffer[dstIndex++] = (char)('0' + (precision / 10) % 10);
-                            precision = (byte)(precision % 10);
-                        }
-
-                        pBuffer[dstIndex++] = (char)('0' + precision);
+                        destination[count++] = (char)('0' + (precision / 100) % 10);
+                        precision = (byte)(precision % 100);
                     }
-                }
 
-                Debug.Assert(dstIndex <= MaxLength);
+                    if (precision >= 10)
+                    {
+                        destination[count++] = (char)('0' + (precision / 10) % 10);
+                        precision = (byte)(precision % 10);
```

Comment:
Maybe we can use DivRem here instead of %.

Reply:
With the consts here, I see better code produced with the % rather than with the DivRem call.

Reply:
I am seeing much smaller disassembly being produced if we used custom DivMod (instead of Math.DivRem):
https://www.diffchecker.com/XMkKfwm9

```C#
public int M1(int precision, Span<char> destination)
{
    int count = 1;
    if (precision >= 10)
    {
        destination[1] = (char)('0' + (precision / 10) % 10);
        count = 2;
        precision = (byte)(precision % 10);
    }
    return count + precision;
}

public int M2(int precision, Span<char> destination)
{
    int count = 1;
    if (precision >= 10)
    {
        uint quotient = DivMod((uint)precision, 10, out uint remainder);
        destination[1] = (char)('0' + quotient % 10);
        count = 2;
        precision = (byte)remainder;
    }
    return count + precision;
}

[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static uint DivMod(uint numerator, uint denominator, out uint modulo)
{
    uint div = numerator / denominator;
    modulo = numerator - (div * denominator);
    return div;
}
```

Also, even with using Math.DivRem, we save an instruction (with microbenchmark showing 20% improvement):
https://www.diffchecker.com/erlKi8ZM

Reply:
Are you sure you're not just seeing the difference between / and % on int vs uint?

I see zero difference in Format when using / and % vs when using either Math.DivRem(uint, uint, out uint) or a custom DivRem that's identical.  In both cases Format clocks in at 136 bytes.

Reply:
> Are you sure you're not just seeing the difference between / and % on int vs uint?

Yep, it's because of using uint instead of int (though the custom DivMod still performs a bit better). In that case, should we cast precision to uint before doing the operations?

Something like:
```C#
public int M0(int precision, Span<char> destination)
{
    int count = 1;
    if (precision >= 10)
    {
        destination[1] = (char)('0' + ((uint)precision / 10) % 10);
        count = 2;
        precision = (byte)((uint)precision % 10);
    }
    return count + precision;
}
```

> Math.DivRem(uint, uint, out uint)

We don't have a Math.DivRem that accepts uint.

|                         Method |      Mean |     Error |    StdDev | Ratio | RatioSD |
|------------------------------- |---------:|----------:|----------:|------:|--------:|
|           WritePrecisionModulo | 874.7 ns | 17.324 ns | 17.791 ns |  1.00 |    0.00 |
|       WritePrecisionModuloUInt | 833.9 ns | 15.611 ns | 13.839 ns |  0.95 |    0.02 |
| WritePrecisionCustomDivModUInt | 791.2 ns |  6.334 ns |  5.289 ns |  0.90 |    0.02 |
|  WritePrecisionCustomDivModInt | 844.5 ns |  9.984 ns |  8.337 ns |  0.96 |    0.02 |
|       WritePrecisionMathDivRem | 898.8 ns | 15.745 ns | 14.728 ns |  1.03 |    0.03 |

Reply:
> In that case, should we cast precision to uint before doing the operations?

It's already declared as `uint precision`.

> We don't have a Math.DivRem that accepts uint.

We do, it's just internal.  This code is in corelib.
https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/System.Private.CoreLib/shared/System/Math.cs#L201

Reply:
> It's already declared as `uint precision`.

Sorry, I missed that. Looks good.